cluster/charts/rook-ceph/templates/resources.yaml

```
@@ -3160,6 +3160,10 @@ spec:
                           crushDeviceClass:
                             description: CrushDeviceClass represents the crush device class for an OSD
                             type: string
+                          crushInitialWeight:
+                            description: CrushInitialWeight represents initial OSD weight in TiB units
+                            pattern: ^([0-9]*[.])?[0-9]+(Ti[B])$
```

What about not including the TiB suffix? Then no need to strip it later. If they could specify different units it could make sense to have the suffix, but if TiB is the only suffix allowed, seems simpler to just document the expected units, and just have the number value.
```suggestion
                            pattern: ^([0-9]*[.])?[0-9]+$
```

---

Ceph expects weight as TiB units. We use those kind of suffix all around, so I think it make sense here as well.

---

Right, it is expected to be in TiB units, but it looks like Ceph doesn't require or allow the units in the value? For example, in [the Ceph doc](https://docs.ceph.com/en/latest/rados/operations/crush-map/), I see an example weight of `2.0`, rather than `2.0TiB`

---

Indeed. And for that reason, I strip the suffix (if exists) before passing it on to Ceph.

---

I'm proposing to match the Ceph doc, which documents the units, then only the number is specified in the value. I don't see why the `TiB` suffix needs to be allowed, the code is simpler without it.

---

OK, will fix